subsys/net/lib/config/init.c

```diff
@@ -278,6 +278,100 @@ static void setup_ipv6(struct net_if *iface, u32_t flags)
 #define setup_ipv6(...)
 #endif /* CONFIG_NET_IPV6 */
 
+static s8_t global_status;
+
+struct l4_event {
+	struct k_work work;
+	struct net_if *iface;
+};
+
+static void send_ready(struct k_work *work)
+{
+	struct l4_event *l4 = CONTAINER_OF(work, struct l4_event, work);
+
+	net_mgmt_event_notify(NET_EVENT_L4_READY, l4->iface);
+}
+
+static void send_disconnected(struct k_work *work)
+{
+	struct l4_event *l4 = CONTAINER_OF(work, struct l4_event, work);
+
+	net_mgmt_event_notify(NET_EVENT_L4_DISCONNECTED, l4->iface);
+}
+
+static void send_event(struct net_if *iface, k_work_handler_t handler)
+{
+	static struct l4_event event;
+
+	k_work_init(&event.work, handler);
+
+	event.iface = iface;
+
+	k_work_submit(&event.work);
+}
+
+static void l4_event_handler(struct net_mgmt_event_callback *cb,
+			     u32_t mgmt_event, struct net_if *iface)
+{
+	if (mgmt_event == NET_EVENT_IPV6_READY ||
+	    mgmt_event == NET_EVENT_IPV4_READY) {
+		send_event(iface, send_ready);
+		global_status++;
+		return;
+	}
+
+	if (mgmt_event == NET_EVENT_IPV6_DISCONNECTED ||
+	    mgmt_event == NET_EVENT_IPV4_DISCONNECTED) {
+		global_status--;
+		if (global_status <= 0) {
+			global_status = 0;
+
+			send_event(iface, send_disconnected);
+		}
+
+		return;
+	}
+
+	if (mgmt_event == NET_EVENT_IF_DOWN) {
+		send_event(iface, send_disconnected);
```

Comment:
Does the application get at least two send_disconnected events, one (or two) from NET_EVENT_IPVx_DISCONNECTED above, and one from if down here. So at least two events, or three if both IPv4 and IPv6 is used. Is this the intention?